File: go/problems/day11_test.go
package problems

import "testing"

var day11 Solvable[int64] = Day11{}

func TestDay11_TestSet_ShouldYield_Result(t *testing.T) {
	input := readProblemInput("day11_1")
	result := day11.Solve(input)

	var expected int64 = 374
	if result != expected {
		t.Errorf("expected %d but got %d", expected, result)
	}
}

func TestDay11_FullSet_ShouldYield_Result(t *testing.T) {
	input := readProblemInput("day11_2")
	result := day11.Solve(input)

	var expected int64 = 10292708
	if result != expected {
		t.Errorf("expected %d but got %d", expected, result)
	}
}

File: ts/problems/day11.test.ts
import test, { describe } from "node:test";
import assert from "node:assert";

import solve from "./day11.js";
import { readProblemInput } from "../inputReader.js";

describe("Day 11", () => {
  test("test set should yield expected result", () => {
    const input = readProblemInput("day11_1");
    const result = solve(input);
    assert.equal(result, 374);
  });

  test("full set should yield expected result", () => {
    const input = readProblemInput("day11_2");
    const result = solve(input);
    assert.equal(result, 10292708);
  });
});
